packages/tx/src/legacyTransaction.ts

```diff
@@ -136,6 +136,18 @@ export default class Transaction extends BaseTransaction<Transaction> {
       }
     }
 
+    if (!this.supports(Capability.EIP155ReplayProtection)) {
+      // In case of a signed non-EIP155 tx, `ecrecover` does not get a `chainId` provided
+      // Internally, `ecrecover` will calculate the recovery bit by `v = v - 27`
+      // This recovery bit is either 0 or 1. Thus, any other value than `v = 27` or `v = 28` will
+      // throw upon calling `getSenderAddress` here. Thus throw early in the constructor
+      if (this.v !== undefined) {
+        if (!this.v.eqn(27) && !this.v.eqn(28)) {
+          throw new Error(`Non-EIP155 txs need either v = 27 or v = 28, got v = ${this.v}`)
+        }
+      }
+    }
```

Comment:
I wouldn't tie this to "after the _validateTxV() analysis" but rather do before respectively directly in the method, think this will get more robust (less ocassion for the analysis to go wrong).

Will try this and update the tests.